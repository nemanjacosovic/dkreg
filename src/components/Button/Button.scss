@import "src/constants/colors.module";

.dkreg-form-controls {
  display: flex;
  margin-top: 30px;
  justify-content: space-between;
}

.dkreg-form-button {
  @media screen and (min-width : 768px) {
    padding: 10px 60px;
  }

  z-index: 500;
  font-size: 13px;
  font-weight: 600;
  overflow: hidden;
  padding: 10px 30px;
  position: relative;
  border-radius: 6px;
  transition: all ease-out 100ms;

  &:focus {
    outline: none;
  }

  &__primary {
    color: $color-white;
    background: $color-primary;
    box-shadow: 0 10px 15px -3px rgba(0, 0, 0, 0.1), 0 4px 6px -2px rgba(0, 0, 0, 0.05);

    &:after {
      top: 0;
      width: 0;
      left: 50%;
      z-index: -2;
      content: '';
      height: 100%;
      position: absolute;
      transition: all ease-in-out 300ms;
      transform: translateY(0) translateX(-50%) skewX(30deg);
      background: $color-primary-hover-mid;
    }

    &:before {
      top: 0;
      left: 50%;
      z-index: -1;
      content: '';
      width: 0;
      height: 100%;
      position: absolute;
      transition-delay: 150ms;
      transition: all ease-in-out 600ms;
      transform: translateY(0) translateX(-50%) skewX(30deg);
      background: $color-primary-hover;
    }

    &.loading {
      cursor: wait;

      & .lds-ellipsis {
        animation: scale-in 300ms ease-in forwards;
      }
    }

    &:hover,
    &.loading {
      &:after,
      &:before {
        width: 150%;
      }
    }
  }

  &__secondary {}

  &__outline {
    font-weight: 400;
    color: $color-link;
    background: $color-white;

    &:hover {
      color: $color-link;
      background: $color-white;
      box-shadow: inset 0 0 0 2px $color-primary;
    }
  }
}

@keyframes scale-in {
  0% {
    opacity: 0;
    transform: scale(0) rotate(-45deg);
  }
  100% {
    opacity: 1;
    transform: scale(0.64) rotate(0deg);
  }
}
