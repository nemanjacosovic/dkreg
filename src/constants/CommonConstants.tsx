export enum ButtonType {
    BUTTON = 'button',
    SUBMIT = 'submit',
    RESET = 'reset'
}

export enum ButtonStyle {
    PRIMARY = 'primary',
    SECONDARY = 'secondary',
    OUTLINE = 'outline'
}

const InputConstants = {
    PASSWORD: 'password',
    TEXT: 'text',
    EMAIL: 'email',
    ICON_SIZE: 22,
    ERROR_TEXT: 'Forkert værdi',
    ERROR_TEXT_REQUIRED: 'Påkrævet område',
    ERROR_TEXT_LENGTH: 'Kontroller længden'
}

export const LanguageConstant = {
    NAME_FIRST: 'Fornavn',
    NAME_FIRST_HINT: '',
    NAME_FIRST_ERROR: InputConstants.ERROR_TEXT,
    NAME_FIRST_ERROR_REQUIRED: InputConstants.ERROR_TEXT_REQUIRED,
    NAME_FIRST_ERROR_LENGTH: InputConstants.ERROR_TEXT_LENGTH,
    NAME_FIRST_PLACEHOLDER: 'Harald',

    NAME_LAST: 'Efternavn',
    NAME_LAST_HINT: '',
    NAME_LAST_ERROR: InputConstants.ERROR_TEXT,
    NAME_LAST_ERROR_REQUIRED: InputConstants.ERROR_TEXT_REQUIRED,
    NAME_LAST_ERROR_LENGTH: InputConstants.ERROR_TEXT_LENGTH,
    NAME_LAST_PLACEHOLDER: 'Holm',

    CITY: 'By',
    CITY_HINT: '2200',
    CITY_ERROR: InputConstants.ERROR_TEXT,
    CITY_ERROR_REQUIRED: InputConstants.ERROR_TEXT_REQUIRED,
    CITY_ERROR_LENGTH: InputConstants.ERROR_TEXT_LENGTH,
    CITY_PLACEHOLDER: 'København',

    POSTAL_CODE: 'Postnummer',
    POSTAL_CODE_HINT: '',
    POSTAL_CODE_ERROR: 'Postnummer',
    POSTAL_CODE_ERROR_REQUIRED: 'Postnummer',
    POSTAL_CODE_ERROR_LENGTH: 'Postnummer',
    POSTAL_CODE_PLACEHOLDER: 'Postnummer',

    EMAIL: 'E-mail',
    EMAIL_HINT: 'Du kan bruge bogstaver, tal og punktum',
    EMAIL_ERROR: InputConstants.ERROR_TEXT,
    EMAIL_ERROR_REQUIRED: InputConstants.ERROR_TEXT_REQUIRED,
    EMAIL_ERROR_LENGTH: InputConstants.ERROR_TEXT_LENGTH,
    EMAIL_PLACEHOLDER: 'henry.holm@email.dk',

    PASSWORD: 'Kodeord',
    PASSWORD_HINT: 'Brug 10 eller flere tegn med en blanding af bogstaver, tal og symboler',
    PASSWORD_ERROR: InputConstants.ERROR_TEXT,
    PASSWORD_ERROR_REQUIRED: InputConstants.ERROR_TEXT_REQUIRED,
    PASSWORD_ERROR_LENGTH: InputConstants.ERROR_TEXT_LENGTH,
    PASSWORD_PLACEHOLDER: 'Vær unik',

    PASSWORD_REPEAT: 'Gentag kodeord',

    DATE_OF_BIRTH: 'Fødselsdato',

    SIGN_UP: 'Tilmelde',

    FORM_RESET: 'Nulstil',
    FORM_SUBMIT: 'Opret profil',

    GREAT_WORK: 'Fantastisk arbejde',
    CHECK_YOUR_INBOX: 'se din indbakke'
}

export default InputConstants;
