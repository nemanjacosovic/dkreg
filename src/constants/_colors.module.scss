$color-primary: #F86E6C;
$color-primary-hover-mid: #F66462;
$color-primary-hover: #F14F4D;
$color-secondary: #009FB7;
$color-alfa: #FED766;
$color-beta: #f0eff3;
$color-gama: #F4F4F8;
$color-white: #FFFFFF;
$color-black: #000000;
$color-gray-40: #666666;
$color-gray-50: #808080;
$color-link: #B23836;
$color-alert: #B20300;

$borderW: 2px;
$numOfBlobs: 8;
