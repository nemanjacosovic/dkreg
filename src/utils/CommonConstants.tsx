const LanguageConstant = {
    NAME_FIRST: 'Fornavn',
    NAME_FIRST_PLACEHOLDER: 'Harald',
    NAME_LAST: 'Efternavn',
    NAME_LAST_PLACEHOLDER: 'Holm',
    EMAIL: 'E-mail',
    EMAIL_PLACEHOLDER: 'henry.holm@email.dk',
    PASSWORD: 'Kodeord',
    PASSWORD_REPEAT: 'Gentag kodeord',
    DATE_OF_BIRTH: 'FÃ¸dselsdato',
    CREATE_ACCOUNT: 'Opret profil',
}

export default LanguageConstant;
