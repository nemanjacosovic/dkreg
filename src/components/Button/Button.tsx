import React, { forwardRef } from 'react';
import { ButtonStyle } from "../../constants/CommonConstants";

// Link, Loader

import './Button.scss';

interface ButtonProps {
    className?: string;
    btnStyle: ButtonStyle;
    url?: string;
    onClick?: React.MouseEventHandler;
    disabled?: boolean;
    toggleContent?: boolean;
    type?: 'button' | 'submit' | 'reset';
    label: string;
    loading?: boolean;
}

type Props = ButtonProps;

export const Button = forwardRef<HTMLButtonElement, Props>((props, ref) => {
    const {
        label,
        onClick,
        type,
        btnStyle
    } = props;


    const _onClick = (event: React.MouseEvent) => {
        if (typeof onClick === 'function') {
            onClick(event);
        }
    };

    const parseButtonStyle = () => {
        switch (btnStyle) {
            case ButtonStyle.PRIMARY:
                return ` dkreg-form-button__${ButtonStyle.PRIMARY}`
            case ButtonStyle.SECONDARY:
                return ` dkreg-form-button__${ButtonStyle.SECONDARY}`
            case ButtonStyle.OUTLINE:
                return ` dkreg-form-button__${ButtonStyle.OUTLINE}`
            default:
                return null;
        }
    };

    return (
        <button type={type} className={`dkreg-form-button${parseButtonStyle()}`} onClick={_onClick}>
            <span>{label}</span>
        </button>
    );
});

export default Button;

