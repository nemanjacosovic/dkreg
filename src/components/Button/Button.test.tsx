import React from 'react';
import { shallow } from 'enzyme';
import Button from './Button';
import {ButtonStyle, ButtonType, LanguageConstant} from "../../constants/CommonConstants";

const testProps = {
    btnStyle: ButtonStyle.PRIMARY,
    type: ButtonType.SUBMIT,
    label: LanguageConstant.FORM_SUBMIT
}

describe('<Button/> component', () => {
    let component = shallow(<Button {...testProps}/>);
    expect(component).toMatchSnapshot();

    it('<Button/> should exist', () => {
        expect(component.instance()).toBeDefined();
        expect(component).toHaveLength(1);
    });

    it('<Button/> should have css classes', () => {
        expect(component.find('button').length).toBe(1);
        expect(component.find('span').length).toBe(1);
        expect(component.find('.dkreg-form-button').length).toBe(1);
        expect(component.find('.dkreg-form-button__primary').length).toBe(1);
    });

    describe('rendering <Button/> as secondary', () => {
        beforeEach(() => {
            const testPropsNew = { ...testProps, btnStyle: ButtonStyle.SECONDARY };
            component = shallow(<Button {...testPropsNew}/>);
        });

        it('<Button/> should exist', () => {
            expect(component.instance()).toBeDefined();
            expect(component).toHaveLength(1);
        });

        it('<Button/> should have css classes', () => {
            expect(component.find('button').length).toBe(1);
            expect(component.find('span').length).toBe(1);
            expect(component.find('.dkreg-form-button').length).toBe(1);
            expect(component.find('.dkreg-form-button__secondary').length).toBe(1);
        });

        it('<Button/> should have correct value', () => {
            expect(component.find('span').text()).toEqual(LanguageConstant.FORM_SUBMIT);
        });
    });

    describe('rendering <Button/> as outline', () => {
        beforeEach(() => {
            const testPropsOut = { ...testProps, btnStyle: ButtonStyle.OUTLINE, label: LanguageConstant.FORM_RESET };
            component = shallow(<Button {...testPropsOut}/>);
        });

        it('<Button/> should exist', () => {
            expect(component.instance()).toBeDefined();
            expect(component).not.toHaveLength(0);
        });

        it('<Button/> should have correct value', () => {
            expect(component.find('span').text()).toEqual(LanguageConstant.FORM_RESET);
        });

        it('<Button/> should have css classes', () => {
            expect(component.find('button').length).toBe(1);
            expect(component.find('span').length).toBe(1);
            expect(component.find('.dkreg-form-button').length).toBe(1);
            expect(component.find('.dkreg-form-button__outline').length).toBe(1);
        });
    });
})
