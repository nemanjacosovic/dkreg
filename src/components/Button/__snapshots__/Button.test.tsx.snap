// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<button
  className="dkreg-form-button dkreg-form-button__primary"
  onClick={[Function]}
  type="submit"
>
  <span>
    Opret profil
  </span>
</button>
`;
